src/algorithms/contractions/ctmrg_contractions.jl

```diff
@@ -1168,13 +1168,8 @@ end
 function renormalize_north_edge(
     E_north::CTMRG_PEPS_EdgeTensor, P_left, P_right, A::PEPSSandwich
```

Comment:
I think with this new implementation we can actually remove the type restrictions as well, and have this as a catch-all fallback. The idea is that this does not access any of the internals of the arguments, nor have any special requirements, so I think this is a good definition. (also for the other ones)
```suggestion
    E_north, P_left, P_right, A```